src/models/pet_steady.js

```diff
@@ -62,3 +62,228 @@ function sweat_rate(t_body) {
+ * @param {number} p_atm - atmospheric pressure, default value 1013.25 [hPa]
+ * @param {number} age - age in years. default=23
+ * @param {number} position - position of the individual (1=sitting, 2=standing, 3=standing, forced convection)
+ * @param {number} sex - default 1. male (1) or female (2).
+ * @param {number} weight - default 75. body mass, [kg]
+ * @param {number} height - default 1.8. height, [m]
+ * @param {number} wme - external work, [W/(m2)] default 0
+ * @returns {number | number[]} - PET (scalar) or energy balance (calculated core, skin, and clo temperatures).
+ */
+function solve_pet(
+  t_arr,
+  _tdb,
+  _tr,
+  _v = 0.1,
+  _rh = 50,
+  _met = 80,
+  _clo = 0.9,
+  actual_environment = false,
+  p_atm = 1013.25,
+  position = 1,
+  age = 23,
+  sex = 1,
+  weight = 75,
+  height = 1.8,
+  wme = 0,
+) {
+  // Constants
+  const e_skin = 0.99; // Skin emissivity
+  const e_clo = 0.95; // Clothing emissivity
+  const h_vap = 2.42 * Math.pow(10, 6); // Latent heat of evaporation [J/Kg]
+  const sbc = 5.67 * Math.pow(10, -8); // Stefan-Boltzmann constant [W/(m2*K^(-4))]
+  const cb = 3640; // Blood specific heat [J/kg/k]
+
+  // Reshape t_arr for fsolve
+  t_arr = [[t_arr[0]], [t_arr[1]], [t_arr[2]]]; // np.reshape(t_arr, (3, 1))
+
+  // Initialize e_bal_vec and other variables
+  const e_bal_vec = [[0], [0], [0]]; // required for the vectorial expression of the balance
+  const a_dubois = body_surface_area(weight, height);
+  //Base metabolism for men and women in [W]
+  const met_female =
+    3.19 *
+    Math.pow(weight, 0.75) *
+    (1.0 +
+      0.004 * (30.0 - age) +
+      0.018 * ((height * 100.0) / Math.pow(weight, 1.0 / 3.0) - 42.1));
+  const met_male =
+    3.45 *
+    Math.pow(weight, 0.75) *
+    (1.0 +
+      0.004 * (30.0 - age) +
+      0.01 * ((height * 100.0) / Math.pow(weight, 1.0 / 3.0) - 43.4));
+  //Attribution of internal energy depending on the sex of the subject
+  const met_correction = sex === 1 ? met_male : met_female;
+  // Source term : metabolic activity
+  const he = (_met + met_correction) / a_dubois;
+  // impact of efficiency
+  const h = he * (1.0 - wme); // [W/m2]
+  // correction for wind
+  const i_m = 0.38; // Woodcock ratio for vapour transfer through clothing [-]
+
+  // Calculation of the Burton surface increase coefficient, k = 0.31 for Hoeppe:
+  const fcl = 1 + 0.31 * _clo; // Increase heat exchange surface depending on clothing level
+  const f_a_cl =
+    (173.51 * _clo - 2.36 - 100.76 * _clo * _clo + 19.28 * _clo ** 3.0) / 100;
+  const a_clo = a_dubois * f_a_cl + a_dubois * (fcl - 1.0); // clothed body surface area
+
+  const f_eff = position === 2 ? 0.696 : 0.725; // effective radiation factor
+
+  const a_r_eff = a_dubois * f_eff; // Effective radiative area depending on the position of the subject
+
+  // Partial pressure of water in the air
+  let vpa = ((_rh / 100.0) * p_sat(_tdb)) / 100; // [hPa]
```

Comment:
can we invert this, as in assign = 12 and if `actual_environment` then do this calculation since this is more expensive than the assignment and we should avoid it if we don't need it